### DISCUSSION TITLE
Unable to use secret in yaml file

### DISCUSSION THREAD

**Discussion by AngeloVilllasanta** [gocd/gocd#13252] [Q&A] [answered]
I have installed the File based secrets plugin for GOCD, and I have added in the json file as such:

![Screenshot from 2024-11-12 11-00-52](https://github.com/user-attachments/assets/123106af-418c-49a2-8eba-7aafad9a76e0)

And in my yaml file, I added it in my secure variables as such:
>   secure_variables: &default_secure_variables
>     PASSWORDFROMVAULT: "some_text_here"
>     SECRET_KEY_1: "{{SECRET:[secret_config_id_in_screenshot][SECRET_KEY_1]}}"
>     SECRET_KEY_2: "{{SECRET:[secret_config_id_in_screenshot][SECRET_KEY_2]}}"
>     SECRET_KEY_3: "{{SECRET:[secret_config_id_in_screenshot][SECRET_KEY_3]}}"

However, I am getting this error:

>INVALID MERGED CONFIGURATION
>Number of errors: 3+
>1. Encrypted value for variable named 'SECRET_KEY_1' is invalid. This usually happens when the cipher text is modified to have an invalid value.
>2. Encrypted value for variable named 'SECRET_KEY_2' is invalid. This usually happens when the cipher text is modified to have an invalid value.
>3. Encrypted value for variable named 'SECRET_KEY_3' is invalid. This usually happens when the cipher text is modified to have an invalid value.

I'm sure that the secret.json file has been added correctly, however, there seems to be an error with how I'm using/fetching the secret keys with my yaml file.

Not sure what I'm missing here, any leads/help would be very much appreciated.

Thank you.

**Comment by chadlwilson** [comment, marked as answer]
Move the values referencing secrets individually to the regular unencrypted `environment_variables` section if you are using secrets resolution, as the references themselves are not secret.

Everything in `secure_variables` is expected to be encrypted with the server-wide encryption key (e.g `AES:fjfjdjfs...`), as an alternative - not resolved from an external secrets store.

**Reply by AngeloVilllasanta** [reply]
This works. Thank you.
